tests/components/test_smoothing.cpp

```diff
@@ -29,15 +35,15 @@ TEST_CASE("smoothing_Newton_Method")
         scheduler.add_operation_type<operations::tri_mesh::VertexSmoothUsingDifferentiableEnergy>(
             "optimize_vertices",
             std::move(op_settings));
-    Tuple tuple = mesh.face_tuple_from_vids(2, 4, 5);
-    spdlog::warn("Initial valuenorm: {}", factory.settings().energy->get_one_ring_value(tuple));
+    Tuple tuple = mesh.face_tuple_from_vids(0, 1, 0);
+    spdlog::warn("Initial valuenorm: {}", factory.settings().energy->get_value(tuple));
     spdlog::warn(
         "Initial gradient: norm: {}",
-        factory.settings().energy->get_local_gradient(tuple).norm());
-    while (factory.settings().energy->get_local_gradient(tuple).norm() > 1e-10) {
+        factory.settings().energy->get_gradient(tuple).norm());
+    while (factory.settings().energy->get_gradient(tuple).norm() > 1e-10) {
```

Comment:
I'd prefer a `for`-loop here with a sufficiently large number of iterations. With a `while`-loop, if your code does not work as expected you will create an endless loop in the tests.